tests/std/tests/LWG3018_shared_ptr_function/test.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory>
+
+using namespace std;
+
+int val = 0;
+
+struct decrement_val {
+    void operator()(void (*const)(int)) const {

```

Comment:
Despite our proclivity for spamming `const`s everywhere, we don't normally put top-level `const` on unnamed function parameters.
```suggestion
    void operator()(void (*)(int)) const {
```

Reply:
I probably have mislead Adam then. Any rationale for special case?

Reply:
Ah, you mean unnamed parameters (of a function) generally, not unnamed parameters of function type. Then got it.

Reply:
The rationale for top-level `const` is to ensure that we don't inadvertently change a value we didn't mean to. Without a name, there's no way to make such a change - inadvertent or otherwise - nor observe a change if it was possible, so the `const` becomes simply noise in the declaration. We therefore omit them.